### ISSUE TITLE
Missing variable initialization in AddToWishlist.php

### ISSUE THREAD

**Issue by alexdpunkt** [magento/magento2#12142]
### Preconditions
1. Magento CE 2.1.10
2. PHP 7.0.22

### Steps to reproduce
Nothing to reproduce. Look at the file `vendor/magento/module-wishlist/Block/AddToWishlist.php`. 
Around line 60, data is pushed to the `$productTypes` variable which has never been initialized before. Result is a PHP error. 

Fix: define `$productTypes` variable before the loop.
```
$productTypes = [];
foreach ($productCollection as $product) {
    $productTypes[] = $product->getTypeId();
}
```

### Expected result
The code should execute without an error. 

### Actual result
PHP exits with an error.

**Comment by RomanKrut** [comment]
@adick agree with you. On 2.2 and 2.3-dev this problem does not exist. Should be fixed in 2.1 version.

**Comment by magento-engcom-team** [comment]
@adick, thank you for your report.
The issue is already fixed in develop branch, 2.2.1
